### PR TITLE
RD2: Elevate Integration Support (Part 1)

### PR DESCRIPTION
**Pull request by force2b**
- As an Admin of an Org with Payment Services, I want new Opportunities created by the Elevate integration to be linked to the proper Recurring Donation if the Opportunity has a CommitmentId specified, but does not have a RecurringDonation Id specified.
- As an Admin of a Payment Services enabled Org, I want to prevent Enhanced Recurring Donations from automatically creating any Installment Opportunities for Elevate-Originating Recurring Donations.
- As an Admin of a Payment Services enabled org, I want to ensure that new Recurring Donations created by the integration are automatically linked with pre-existing Opportunities that have the same Elevate Commitment Id as the Recurring Donation.
- Ensure that Recurring Donations created by the Payment Services integration are never allowed to be updated manually by a User in NPSP other than through the Payment Services Integration User, with the exception of rollup and custom fields, so that I can be sure that the Recurring Donation information in NPSP exactly matches what is in Elevate.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

Apex Classes:
* PS_IntegrationServiceConfig
* PS_IntegrationServiceConfig_TEST
* RD2_ElevateIntegrationService
* RD2_ElevateIntegrationService_TEST

Custom Field:
* Opportunity.CommitmentId__c

Custom Label:
- RD2_ElevatePermissionRequired

# Deleted Metadata

# Unaggregated Pull Requests
* RD2:  Do not auto-create Opportunities for Elevate originated RDs [[PR5351](https://github.com/SalesforceFoundation/NPSP/pull/5351)]
* RD2: Find matching Opps by CommitmentId [[PR5333](https://github.com/SalesforceFoundation/NPSP/pull/5333)]
*  RD2: Link Opportunity To RD Through CommitmentId [[PR5366](https://github.com/SalesforceFoundation/NPSP/pull/5366)]
* RD2: Prevent users from changing Elevate originated RDs [[PR5338](https://github.com/SalesforceFoundation/NPSP/pull/5338)]

### CODE REVIEW

File: src/objects/Opportunity.object [hunk 1]
```diff
@@ -844,6 +844,19 @@
         <type>LongTextArea</type>
         <visibleLines>3</visibleLines>
     </fields>
+    <fields>
+        <fullName>CommitmentId__c</fullName>
```
Comment (ptewson-sfdo) [inline src/objects/Opportunity.object:848]: @melissabarber have you had a chance to look at these fields and align with the rest of your efforts?

File: src/classes/ERR_Handler_API.cls [hunk 2]
```diff
@@ -68,6 +68,8 @@ global class ERR_Handler_API {
         REL,
         /* @description Recurring Donations context. */
         RD,
+        /* @description Elevate Integration context. */
+        Elevate,
```
Comment (ptewson-sfdo) [inline src/classes/ERR_Handler_API.cls:72]: @psadie-sfdo  just FYI that this is coming in from the Recurring donations side of things

File: src/classes/PS_IntegrationServiceConfig.cls [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+*/
+public with sharing class PS_IntegrationServiceConfig {
+
+    /**
+    * @description Determines if the Elevate integration service is configured and enabled.
+    * The Elevate Payment Services integration is considered "enabled" (for now) when there
+    * is at least a single record in the protected Payment Services Configuration object
+    * @return Boolean
+    */
+    private Boolean isEnabled {
```
Comment (ptewson-sfdo) [inline src/classes/PS_IntegrationServiceConfig.cls:44]: @kenlewis-sfdo FYI for possible coordination

Comment (force2b) [reply src/classes/PS_IntegrationServiceConfig.cls:44]: Thanks @ptewson-sfdo. This is hardcoded for now, so none of our new Elevate support logic will run until this method and the `hasIntegrationPermissions` method are updated to actually detect that the Elevate Integration is enabled. Not sure when that logic will be available. My assumption is that @pedrodalcol's team (or possibly your team) will add that logic into these methods when it's ready, but let me know asap if that's not the case.

Comment (ptewson-sfdo) [reply src/classes/PS_IntegrationServiceConfig.cls:44]: Right, @force2b  - @kenlewis-sfdo is delivering that for our team this sprint and I just wanted to make sure this loop was closed. I don't think he had previously been made aware of this stubbed method's planned existence.

Comment (force2b) [reply src/classes/PS_IntegrationServiceConfig.cls:44]: Ah. Makes sense. Thanks!

**Pull request merged**